Automattic/jetpack
# Action Bar: add following, liking, and more options modal

#### Changes proposed in this Pull Request:

<img width="459" alt="image" src="https://user-images.githubusercontent.com/1699996/192343643-e614b8d1-3320-4c4f-8ba0-24913c21cb10.png">


- Adds a working like button for single posts
- Adds a working follow button for the site
- Adds modal with more actions

Combine with D87870-code to test liking and following

Fixes #27356 (dropping a CSS var not available in 2023 theme)

#### Other information:


#### Jetpack product discussion

pe7F0s-1U-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

Apply D87870-code and sandbox the specified domains

- Switch to this branch and build the project: jetpack install --all
- Build the action bar package: jetpack build packages/action-bar
- Activate the Action bar module from /wp-admin/admin.php?page=jetpack_modules
- Open a single post page on the front-end, and you should see the action bar in the lower left corner
- Click on the dots to see more options
- Click on the star to like or unlike the post
- Click on the plus to follow or unfollow the site

You can check likes by visiting https://wordpress.com/activities/likes

You can check follows by visiting https://wordpress.com/following/manage

Additional testing

- With a wpcom sandbox, modify the api to return an error response.
- You should see an error message, like the one below

<img width="487" alt="image" src="https://user-images.githubusercontent.com/1699996/192343975-7f14465e-8c71-4a1f-8cb8-b9e5960fbe29.png">

- Test with third-party cookies disabled (e.g. use a Chrome incognito window)... liking and following should still work

Note there is a known issue if testing with *.jurassic.ninja or *.jurassic.tube that follows the parent site instead of the subdomain, see D88262-code.
